File: crates-archive/superconfig-ffi/reference/test_usage.js
#!/usr/bin/env node
/**
 * Test script to demonstrate using SuperConfig from Node.js
 */

try {
  const { SuperConfig } = require("./superconfig_ffi.node");

  console.log("✅ Successfully imported SuperConfig from Node.js module!");

  // Create a new SuperConfig instance
  const config = new SuperConfig();
  console.log("✅ Created SuperConfig instance:", config);

  // Test getting verbosity
  const verbosity = config.getVerbosity();
  console.log("✅ Default verbosity:", verbosity);

  // Test with debug verbosity
  const debugConfig = config.withDebugVerbosity();
  const debugVerbosity = debugConfig.getVerbosity();
  console.log("✅ Debug verbosity:", debugVerbosity);

  // Test with trace verbosity
  const traceConfig = config.withTraceVerbosity();
  const traceVerbosity = traceConfig.getVerbosity();
  console.log("✅ Trace verbosity:", traceVerbosity);

  console.log("\n🎉 All Node.js bindings working correctly!");
} catch (error) {
  console.error("❌ Error:", error.message);
  console.error(
    'Make sure to build with: cargo build --features "nodejs" --release',
  );
}

File: crates/superconfig-ffi/reference/test_usage.py
#!/usr/bin/env python3
"""Test script to demonstrate using SuperConfig from Python"""

try:
    import superconfig_ffi
    
    print("✅ Successfully imported superconfig_ffi module!")
    
    # Create a new SuperConfig instance
    config = superconfig_ffi.SuperConfig()
    print(f"✅ Created SuperConfig instance: {config}")
    
    # Test getting verbosity
    verbosity = config.get_verbosity()
    print(f"✅ Default verbosity: {verbosity}")
    
    # Test with debug verbosity
    debug_config = config.with_debug_verbosity()
    debug_verbosity = debug_config.get_verbosity()
    print(f"✅ Debug verbosity: {debug_verbosity}")
    
    # Test with trace verbosity
    trace_config = config.with_trace_verbosity()
    trace_verbosity = trace_config.get_verbosity()
    print(f"✅ Trace verbosity: {trace_verbosity}")
    
    print("\n🎉 All Python bindings working correctly!")
    
except ImportError as e:
    print(f"❌ Failed to import superconfig_ffi: {e}")
except Exception as e:
    print(f"❌ Error using SuperConfig: {e}")